features/netsocket/DNS.h

```diff
@@ -42,6 +42,23 @@ class DNS {
      */
     virtual nsapi_error_t gethostbyname(const char *host,
                                         SocketAddress *address, nsapi_version_t version = NSAPI_UNSPEC, const char *interface_name = NULL) = 0;
+
+    /** Translate a hostname to the multiple IP addresses with specific version using network interface name.
+     *
+     *  The hostname may be either a domain name or an IP address. If the
+     *  hostname is an IP address, no network transactions will be performed.
+     *
+     *  If no stack-specific DNS resolution is provided, the hostname
+     *  will be resolve using a UDP socket on the stack.
+     *
```

Comment:
Please also specify that user should allocate the space for results.

Reply:
And how much, it should allocate.

Reply:
Seppo  getaddrinfo allocates space internally.
Due to  function doesn't know  how many DNS records  will be found it uses predefined const.
MBED_CONF_NSAPI_DNS_ADDRESSES_LIMIT  set to 10 (instagram.com has 8 IPs  )  but  app json can override it. If number of record is smaller than allocate limit  extra memory is deleted to avoid memory leak